src/editors/sharedComponents/ImageUploadModal/ImageSettingsModal/DimensionControls.test.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect } from 'react';
+import * as paragon from '@openedx/paragon';
+import {
+  fireEvent, render, screen, waitFor, initializeMocks,
+} from '@src/testUtils';
+import DimensionControls from './DimensionControls';
+import * as hooks from './hooks';
+
+const WrappedDimensionControls = () => {
+  const dimensions = hooks.dimensionHooks('altText');
+
+  useEffect(() => {
+    dimensions.onImgLoad({ })({ target: { naturalWidth: 1517, naturalHeight: 803 } });
+  }, []);
+
+  return <DimensionControls {...dimensions} />;
+};
+
+const UnlockedDimensionControls = () => {
+  const dimensions = hooks.dimensionHooks('altText');
+
+  useEffect(() => {
+    dimensions.onImgLoad({ })({ target: { naturalWidth: 1517, naturalHeight: 803 } });
+    dimensions.unlock();
+  }, []);
+
+  return <DimensionControls {...dimensions} />;
+};
+
+describe('DimensionControls', () => {
+  describe('render', () => {
+    const props = {
+      lockAspectRatio: { width: 4, height: 5 },
+      locked: { 'props.locked': 'lockedValue' },
+      isLocked: true,
+      value: { width: '20', height: '40' },
+      setWidth: jest.fn(),
+      setHeight: jest.fn(),
+      lock: jest.fn(),
+      unlock: jest.fn(),
+      updateDimensions: jest.fn(),
+    };
+    beforeEach(() => {
+      jest.spyOn(hooks, 'onInputChange').mockImplementation((handler) => ({ 'hooks.onInputChange': handler }));
+      initializeMocks();
+    });
+    afterEach(() => {
+      jest.spyOn(hooks, 'onInputChange').mockRestore();
+    });
+    test('renders component', () => {
+      render(<DimensionControls {...props} />);
+      expect(screen.getByText('Image Dimensions')).toBeInTheDocument();
+    });
+    test('renders nothing with null value', () => {
+      const reduxProviderWrapper = '<div data-testid="redux-provider"></div>';
+      const { container } = render(<DimensionControls {...props} value={null} />);
+      expect(screen.queryByText('Image Dimensions')).not.toBeInTheDocument();
+      expect(container.innerHTML).toBe(reduxProviderWrapper);
+      expect(container.firstChild?.textContent).toBe('');
+    });
+
+    test('renders locked and unlocked icon button according to isLocked prop', () => {
+      const { rerender } = render(<DimensionControls {...props} isLocked={false} />);
+      expect(screen.getByRole('button', { name: 'lock dimensions' })).toBeInTheDocument();
+      rerender(<DimensionControls {...props} isLocked />);
+      expect(screen.getByRole('button', { name: 'unlock dimensions' })).toBeInTheDocument();
+    });
+  });
+
+  describe('component tests for dimensions', () => {
+    beforeEach(() => {
+      paragon.Form.Control = jest.fn().mockImplementation(({ value, onChange, onBlur }) => (
+        <input className="formControl" onChange={onChange} onBlur={onBlur} value={value} />
+      ));
```

Comment:
Could you please add a comment explaining why this particular mock is useful? We generally don't mock out Paragon components like this in the tests.

I'm also wondering if we can use `jest.SpyOn(paragon.Form, 'Control').mockImplementation(...)` instead of directly overriding it, for better integration with jest and less chance of interfering with other tests in the same module.